### PR TITLE
Adds adversarial example generation script

### PR DESCRIPTION
**Pull request by ellenjiang7**
Adds the adversarial example generating script (in adversarial_depth.py)
which takes in a DepthModelWrapper instance (in depth_model_wrapper.py),
which has methods to load and call predict on a depth model, and moves
adversarial code into a separate directory.

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/depth-maps-art-and-illusions/11)

### CODE REVIEW

File: adversarial/adversarial_depth.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+from matplotlib import pyplot as plt
+from PIL import Image
+
+
+# Image Paths
+SINK_IMAGE = "images/sink.png"
+BATHTUB_IMAGE = "images/bathtub.png"
+
+
+def get_sink_to_bathtub_inputs(depth_model_wrapper, model):
```
Comment (annxingyuan) [inline adversarial/adversarial_depth.py]: Maybe add a brief description of the sink to bathtub experiment?

File: adversarial/adversarial_depth.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+	"""
+  for i in range(iterations):
+      loss, gradients = retrieve_loss_and_gradients(input_image, loss_gradient_function)
+      print('Loss at %d: %.3f' % (i, loss))
+      input_image -= step_size * gradients
+  return input_image
+
+
+def generate_adversarial_example(depth_model_wrapper, iterations = 100,
```
Comment (annxingyuan) [inline adversarial/adversarial_depth.py]: Maybe make input_image and goal_output configurable?

**Comment by EmilyReif** [comment]
___
*[adversarial/adversarial_depth.py, line 26 at r1](https://reviewable.io/reviews/pair-code/depth-maps-art-and-illusions/11#-LlRq6IF6xk_FpmFX9dD:-LlmckXi4vj793i9QVBb:b-sf8yw5) ([raw file](https://github.com/pair-code/depth-maps-art-and-illusions/blob/755bc2d3e2143eb8b67ec507368b12fef48997a1/adversarial/adversarial_depth.py#L26)):*
<details><summary><i>Previously, ellenjiang7 wrote…</i></summary><blockquote>

Hmm it suppresses logging, though now that I think about it - perhaps it might be better to remove this and allow the default logging, in case it's helpful for someone troubleshooting when they use the script?
</blockquote></details>

sure, sgtm

**Comment by EmilyReif** [comment]
___
*[adversarial/depth_model_wrapper.py, line 16 at r1](https://reviewable.io/reviews/pair-code/depth-maps-art-and-illusions/11#-LlRrpAf-TS9VOSdY67F:-Llmd3B3DNBgWwY00sLN:bbshyx1) ([raw file](https://github.com/pair-code/depth-maps-art-and-illusions/blob/755bc2d3e2143eb8b67ec507368b12fef48997a1/adversarial/depth_model_wrapper.py#L16)):*
<details><summary><i>Previously, ellenjiang7 wrote…</i></summary><blockquote>

Yeah, sounds great! Does this work?
</blockquote></details>

Looks perfect!

**Comment by EmilyReif** [comment]
___
*[depth_visualizer/static/escher-trippy.5b69c0dc.jpg, line 0 at r3](https://reviewable.io/reviews/pair-code/depth-maps-art-and-illusions/11#-LlmdUUpDjRD2G1ykcfA:-LlmdUUq5N2zCjdQxE-P:bgxr5m9) ([raw file](https://github.com/pair-code/depth-maps-art-and-illusions/blob/5e7e15df5548d5ac2869bb203a8c4e21f3e598e6/depth_visualizer/static/escher-trippy.5b69c0dc.jpg#L0)):*
wait, I think this is a generated file? Or maybe reviewable is just showing me this cause it was in the PR at one point... (just check to make sure that in your current branch, there's no more generated files)

**Comment by EmilyReif** [comment]
Looks good to me!

**Comment by EmilyReif** [comment]
:lgtm:

**Comment by EmilyReif** [comment]
___
*[depth_visualizer/static/illusions.77de5100.css, line 1 at r1](https://reviewable.io/reviews/pair-code/depth-maps-art-and-illusions/11#-LlRsPP92uhLAkBu7J9b:-LlmeJQiBPqkJaGXSFOC:b-qg7e5t) ([raw file](https://github.com/pair-code/depth-maps-art-and-illusions/blob/755bc2d3e2143eb8b67ec507368b12fef48997a1/depth_visualizer/static/illusions.77de5100.css#L1)):*
<details><summary><i>Previously, ellenjiang7 wrote…</i></summary><blockquote>

Oh I just realized I forgot to change the paths of the static folder items in gitignore when I moved the starter project into a subdirectory. Fixed now.
</blockquote></details>

gotcha, sounds good!

**Pull request merged**